Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.cs

```diff
@@ -0,0 +1,214 @@
+                    {
+                        throw new PlatformNotSupportedException($"Add-IshBackgroundTask with the current parameter set requires server-side BackgroundTask API which only available starting from 13SP2/13.0.2 and up. ServerIshVersion[{IshSession.ServerVersion}]");
+                    }
+                    break;
+                case "IshObjectsGroup":
+                    if ((IshSession.ServerIshVersion.MajorVersion < 14) || ((IshSession.ServerIshVersion.MajorVersion == 14) && (IshSession.ServerIshVersion.RevisionVersion < 4)))
+                    {
+                        throw new PlatformNotSupportedException($"Add-IshBackgroundTask with the current parameter set requires server-side DocumentObj API which only available starting from 14SP4/14.0.4 and up. ServerIshVersion[{IshSession.ServerVersion}]");
+                    }
+                    break;
+            }
+            base.BeginProcessing();
+        }
+
+        /// <summary>
+        /// Process the cmdlet.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                if (IshObject != null)
+                {
+                    foreach (IshObject ishObject in IshObject)
+                    {
+                        _retrievedIshObjects.Add(ishObject);
+                    }
+                }
+            }
+            catch (TrisoftAutomationException trisoftAutomationException)
+            {
+                ThrowTerminatingError(new ErrorRecord(trisoftAutomationException, base.GetType().Name, ErrorCategory.InvalidOperation, null));
+            }
+            catch (Exception exception)
+            {
+                ThrowTerminatingError(new ErrorRecord(exception, base.GetType().Name, ErrorCategory.NotSpecified, null));
+            }
+        }
+
+        /// <summary>
+        /// Process the Add-IshBackgroundTask command-let.
+        /// </summary>
+        /// <exception cref="TrisoftAutomationException"></exception>
+        /// <exception cref="Exception"></exception>
+        /// <remarks>Writes <see cref="IshBackgroundTask"/> to the pipeline.</remarks>
+        protected override void EndProcessing()
+        {
+            try
+            {
+                IshFields metadataFilter = new IshFields();
+                IshFields requestedMetadata = IshSession.IshTypeFieldSetup.ToIshRequestedMetadataFields(IshSession.DefaultRequestedMetadata, ISHType, new IshFields(), Enumerations.ActionMode.Find);
+                long progressId = 0;
+                if (ParameterSetName == "IshObjectsGroup")
+                {
+                    var lngCardIds = _retrievedIshObjects.Select(ishObject => Convert.ToInt64(ishObject.ObjectRef[Enumerations.ReferenceType.Lng])).ToList();
+                    progressId = IshSession.DocumentObj25.RaiseEventByIshLngRefs(lngCardIds.ToArray(), EventType);
```

Comment:
@HildeVermeiren @OlegTokar  ... should we not do batching of BackgroundTasks? Like per 300 batch size... this cmdlet will force a folder with 5000 objects x 10 version x 20 language = 100 000 lngcardids to be passed in ONE batch? What is the max batch size we tested with?
If we do batching then code needs to move from EndProcessing into pipeline friendly ProcessRecord()

Reply:
@ddemeyer The API/business code is just validating the incoming languageCardIds and makes a (big) eventdata xml with it. So if it are 100000 languageCardIds, then I am not 100% that it will work, but I am not sure how you get those 100000 objects from the server to client via ISHRemote neither. The background task will just do them sequential which might also take long. 
I suppose we can split them in batch, but then you need need different events (eventDescription + number) and will get different background tasks ...so it will also change the result and how to retrieve the result.
@OlegTokar I don't think we really did stress testing to this method to find the maximum for one call?

Reply:
@HildeVermeiren  No, we don't know the maximum for the RaiseEventByIshLngRefs

Reply:
Quick meeting... ISHRemote will do client-side grouping of the whole result in EndProcessing()... This will allow to sort all lngcards by logical id (so that the same logical id is not showing up in 2+ backgroundtasks, causing locking issues serverside forcing the full backgroundtask to restart). Use session.MetadataBatchSize (currently 1000, more optimized 999 for Oracle (seperate issue)) for sizing... Event naming should be an increasing number